packages/cli/src/api/e2e.api.ts

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/naming-convention */
+import { Router } from 'express';
+import bodyParser from 'body-parser';
+import { v4 as uuid } from 'uuid';
+import config from '@/config';
+import * as Db from '@/Db';
+import { RoleNames, RoleScopes } from '@/databases/entities/Role';
+import { hashPassword } from '@/UserManagement/UserManagementHelper';
+
+if (process.env.E2E_TESTS !== 'true') {
+	console.error('E2E endpoints only allowed during E2E tests');
+	process.exit(1);
+}
+
+const tablesToTruncate = [
+	'shared_workflow',
+	'shared_credentials',
+	'webhook_entity',
+	'workflows_tags',
+	'credentials_entity',
+	'tag_entity',
+	'workflow_entity',
+	'execution_entity',
+	'settings',
+	'installed_packages',
+	'installed_nodes',
+	'user',
+	'role',
+];
```

Comment:
Can we query for all table names and exclude `migrations` and `sqlite_sequence`?

Reply:
I did exactly that first, but then I couldn't sort the tables in the order foreign key dependencies, and since we don't have `ON DELETE CASCADE`, truncating `role` threw an errors as `user` depended on it.

I already tried setting `PRAGMA foreign_keys=OFF`, tried doing the whole thing in a transaction as well, but none of these worked for some reason. 
If you know another way to disable foreign-key constraints on sqlite, I'd be glad to change this. I don't want any of us having to maintain this list either.

plus, if we can temporarily disable foreign-key constraints, we can fire these queries in parallel, and make this a few miliseconds faster 😅

Reply:
For reference, this works fine in BE tests: https://github.com/n8n-io/n8n/blob/aec08275aa0e7d9f9d0c5fc787743a0ca7476386/packages/cli/test/integration/shared/testDb.ts#L178-L194

Else we can revisit as tech debt.